src/graph-layouts.jl

```diff
@@ -0,0 +1,89 @@
+export distgraph, distdigraph, num_vertices, vertices, adj
+###
+# Graph data structure definitions.
+###
+
+# Allowed Graph datastructures:
+typealias AdjacencyList Array{Array{Int, 1}, 1}
+typealias AdjacencyMatrix AbstractArray{Int, 2}                      # Covers sparse matrices
+typealias GraphStruct Union{AdjacencyList, AdjacencyMatrix}
+
+
+# Generic (undirected or directed, up to the user to decide) Graph
+type DistGraph{S<:GraphStruct}
```

Comment:
why is this needed? `distribute` combined with  `GraphLayout` can do the job right?